### PR TITLE
[MLMD Lineage] Integrate LineageView changes to add navigation to card resources

### PR DESCRIPTION
**Pull request by kwasi**
/assign @kwasi
/cc @avdaredevil 
/area metadata
/area front-end

* Adds changes from kubeflow/frontend to make LineageCardRow titles link to Lineage resource details page
* Pass route builder into LineageView from ArtifactDetails

## Validation

Deployment: https://kubeflow-ui-07.endpoints.kwasinti-kubeflow-dev.cloud.goog/

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/metadata/211)

### CODE REVIEW

**Comment by kwasi** [comment]
/hold

This PR is ready for review but needs https://github.com/kubeflow/frontend/pull/22 to be merged first so one more commit to bump the hash in package.json can be made.

**Comment by avdaredevil** [comment]
https://github.com/kubeflow/frontend/pull/22 is merged, so as soon as you make your commit hash update, I can review this PR :)

**Comment by kwasi** [comment]
Ready for review and pushed new working docker image.

File: frontend/src/pages/ArtifactDetails.tsx [hunk 1]
```diff
@@ -61,6 +62,16 @@ export default class ArtifactDetails extends Page<{}, ArtifactDetailsState> {
     this.load = this.load.bind(this);
   }
 
+  componentDidUpdate(prevProps: Readonly<{} & PageProps>, prevState: Readonly<ArtifactDetailsState>, snapshot?: any): void {
+    if (this.props.match.params[RouteParams.ID] !== prevProps.match.params[RouteParams.ID]) {
```
Comment (avdaredevil) [inline frontend/src/pages/ArtifactDetails.tsx]: _**Nit**: Invert condition to early return (decrease code depth)_
```suggestion
    if (this.props.match.params[RouteParams.ID] === prevProps.match.params[RouteParams.ID]) return
```

File: frontend/src/pages/ArtifactDetails.tsx [hunk 2]
```diff
@@ -153,4 +167,15 @@ export default class ArtifactDetails extends Page<{}, ArtifactDetailsState> {
   private switchTab(selectedTab: number) {
     this.setState({selectedTab});
   }
+
+  private static buildResourceDetailsPageRoute(
+    resource: LineageResource, typeName: string): string {
```
Comment (avdaredevil) [inline frontend/src/pages/ArtifactDetails.tsx]: Is this too big for a line?

File: frontend/src/pages/ArtifactDetails.tsx [hunk 3]
```diff
@@ -153,4 +167,15 @@ export default class ArtifactDetails extends Page<{}, ArtifactDetailsState> {
...
+    let route;
+    if (resource instanceof Artifact) {
+      route = RoutePage.ARTIFACT_DETAILS.replace(`:${RouteParams.ARTIFACT_TYPE}+`, typeName);
+    } else {
+      route = RoutePage.EXECUTION_DETAILS.replace(`:${RouteParams.EXECUTION_TYPE}+`, typeName);
+    }
+    return route.replace(`:${RouteParams.ID}`, String(resource.getId()));
+  }
```
Comment (avdaredevil) [inline frontend/src/pages/ArtifactDetails.tsx]: Just a suggestion for brevity, entire function can be:
```js
private static buildResourceDetailsPageRoute(resource: LineageResource, typeName: string): string {
    const tp = resource instanceof Artifact ? 'ARTIFACT' : 'EXECUTION'
    const route = RoutePage[`${tp}_DETAILS`]
        .replace(`:${RouteParams[`${tp}_TYPE`]}+`, typeName)
        .replace(`:${RouteParams.ID}`, String(resource.getId()));
    return route;
}
```

**Comment by avdaredevil** [comment]
Hey @kwasi 
There are a few comments on this PR. So once you get a chance to go through them, I can review / leave an lgtm.

File: frontend/src/pages/ArtifactDetails.tsx (hunk 2 above)
Comment (kwasi) [reply frontend/src/pages/ArtifactDetails.tsx]: Yes, it's 101 characters.

File: frontend/src/pages/ArtifactDetails.tsx (hunk 1 above)
Comment (kwasi) [reply frontend/src/pages/ArtifactDetails.tsx]: Done

File: frontend/src/pages/ArtifactDetails.tsx (hunk 3 above)
Comment (kwasi) [reply frontend/src/pages/ArtifactDetails.tsx]: Simplified a bit, but I prefer readability over brevity and the ability for tooling to understand what ROUTE_PARAMS.KEY means.

**Comment by kwasi** [comment]
@avdaredevil This PR should be ready to go. I've also updated to the latest hash and updated the deployment.

**Pull request merged**